pydfc/dfc_methods/sliding_window.py

```diff
@@ -141,36 +131,25 @@ def dFC(self, time_series, W=None, n_overlap=None, tapered_window=False):
             model.fit(time_series.T)
             self.graphical_lasso_alpha_ = model.alpha_
 
-        window_taper = signal.windows.gaussian(W, std=3 * W / 22)
-        # C = DFC(measure=self)
         FCSs = list()
         TR_array = list()
         for l in range(0, L - W + 1, step):
 
-            ######### creating a rectangel window ############
+            # Create rectangular window
             window = np.zeros((L))
             window[l : l + W] = 1
 
-            ########### tapering the window ##############
+            # Taper the window
             if tapered_window:
-                window = signal.convolve(window, window_taper, mode="same") / sum(
-                    window_taper
-                )
-
-            window = np.repeat(
-                np.expand_dims(window, axis=0), time_series.shape[0], axis=0
-            )
-
-            # int(l-W/2):int(l+3*W/2) is the nonzero interval after tapering
-            FCSs.append(
-                self.FC(
-                    np.multiply(time_series, window)[
-                        :, max(int(l - W / 2), 0) : min(int(l + 3 * W / 2), L)
-                    ]
-                )
-            )
+                std = window_std if window_std is not None else 3 * W / 22
+                window_taper = signal.windows.gaussian(W, std=std)
+                #window = signal.convolve(window, window_taper, mode="same") / sum(window_taper)
```

Comment:
Please note that the result of the this is different from what is described in Allen et al. paper. A tapered window is a rectangle convolved with a gaussian not a gaussian.